### ISSUE TITLE
The system() function returns weird symbols in the command line after v8.2.0766

### ISSUE THREAD

**Issue by betheev** [vim/vim#6112]
**Bug description**
In Vim v8.2.0803 (but starting with v8.2.0767), the system() function writes weird symbols to the command line.

**How to reproduce**
1. Run `vim -u NONE`
2. Enter the command `:echo system("ls")`
3. The function returns the list of files in the current workspace, as expected, but the result contains weird symbols at the beginning:

[>4;m[>4;2
Desktop
Documents
Downloads
Music
Pictures
Public
Templates
Videos

**Expected behavior**
In the example, the function should return:
Desktop
Documents
Downloads
Music
Pictures
Public
Templates
Videos

**Environment:**
VIM - Vi IMproved 8.2 (2019 Dec 12, compiled May 21 2020 16:32:52)
Included patches: 1-767
Compiled by ******
Huge version with GTK2 GUI.  Features included (+) or not (-):
+acl               -farsi             +mouse_sgr         +tag_binary
+arabic            +file_in_path      -mouse_sysmouse    -tag_old_static
+autocmd           +find_in_path      +mouse_urxvt       -tag_any_white
+autochdir         +float             +mouse_xterm       -tcl
-autoservername    +folding           +multi_byte        +termguicolors
+balloon_eval      -footer            +multi_lang        +terminal
+balloon_eval_term +fork()            -mzscheme          +terminfo
+browse            +gettext           +netbeans_intg     +termresponse
++builtin_terms    -hangul_input      +num64             +textobjects
+byte_offset       +iconv             +packages          +textprop
+channel           +insert_expand     +path_extra        +timers
+cindent           +ipv6              -perl              +title
+clientserver      +job               +persistent_undo   +toolbar
+clipboard         +jumplist          +popupwin          +user_commands
+cmdline_compl     +keymap            +postscript        +vartabs
+cmdline_hist      +lambda            +printer           +vertsplit
+cmdline_info      +langmap           +profile           +virtualedit
+comments          +libcall           -python            +visual
+conceal           +linebreak         +python3           +visualextra
+cryptv            +lispindent        +quickfix          +viminfo
+cscope            +listcmds          +reltime           +vreplace
+cursorbind        +localmap          +rightleft         +wildignore
+cursorshape       -lua               -ruby              +wildmenu
+dialog_con_gui    +menu              +scrollbind        +windows
+diff              +mksession         +signs             +writebackup
+digraphs          +modify_fname      +smartindent       +X11
+dnd               +mouse             -sound             -xfontset
-ebcdic            +mouseshape        +spell             +xim
+emacs_tags        +mouse_dec         +startuptime       -xpm
+eval              -mouse_gpm         +statusline        +xsmp_interact
+ex_extra          -mouse_jsbterm     -sun_workshop      +xterm_clipboard
+extra_search      +mouse_netterm     +syntax            -xterm_save
   system vimrc file: "$VIM/vimrc"
     user vimrc file: "$HOME/.vimrc"
 2nd user vimrc file: "~/.vim/vimrc"
      user exrc file: "$HOME/.exrc"
  system gvimrc file: "$VIM/gvimrc"
    user gvimrc file: "$HOME/.gvimrc"
2nd user gvimrc file: "~/.vim/gvimrc"
       defaults file: "$VIMRUNTIME/defaults.vim"
    system menu file: "$VIMRUNTIME/menu.vim"
  fall-back for $VIM: "/usr/local/share/vim"
Compilation: gcc -c -I. -Iproto -DHAVE_CONFIG_H -DFEAT_GUI_GTK  -pthread -I/usr/include/gtk-2.0 -I/usr/lib/x86_64-linux-gnu/gtk-2.0/include -I/usr/include/gio-unix-2.0/ -I/usr/include/cairo -I/usr/include/pango-1.0 -I/usr/include/atk-1.0 -I/usr/include/cairo -I/usr/include/pixman-1 -I/usr/include/gdk-pixbuf-2.0 -I/usr/include/libpng16 -I/usr/include/pango-1.0 -I/usr/include/harfbuzz -I/usr/include/pango-1.0 -I/usr/include/glib-2.0 -I/usr/lib/x86_64-linux-gnu/glib-2.0/include -I/usr/include/freetype2 -I/usr/include/libpng16 -I/usr/include/freetype2 -I/usr/include/libpng16   -g -O2 -U_FORTIFY_SOURCE -D_FORTIFY_SOURCE=1       
Linking: gcc   -L/usr/local/lib -Wl,--as-needed -o vim   -lgtk-x11-2.0 -lgdk-x11-2.0 -lpangocairo-1.0 -latk-1.0 -lcairo -lgdk_pixbuf-2.0 -lgio-2.0 -lpangoft2-1.0 -lpango-1.0 -lgobject-2.0 -lglib-2.0 -lfontconfig -lfreetype -lSM -lICE -lXt -lX11 -lXdmcp -lSM -lICE  -lm -ltinfo   -lacl -lattr -ldl     -L/usr/lib/python3.6/config-3.6m-x86_64-linux-gnu -lpython3.6m -lpthread -ldl -lutil -lm  

 - OS: Ubuntu 18.04.4 LTS (4.15.0-96-generic #97-Ubuntu SMP Wed Apr 1 03:25:46 UTC 2020 x86_64 x86_64 x86_64 GNU/Linux)

 - Terminal: Xfce Terminal Emulator v0.8.7.4

**Comment by brammool** [comment]
> **Bug description**
> In Vim v8.2.0803 (but starting with v8.2.0767), the system() function writes weird symbols to the command line.
> 
> **How to reproduce**
> 1. Run `vim -u NONE`
> 2. Enter the command `:echo system("ls")`
> 3. The function returns the list of files in the current workspace, as expected, but the result contains weird symbols at the beginning:
> 
> [>4;m[>4;2
> Desktop
> Documents
> Downloads
> Music
> Pictures
> Public
> Templates
> Videos
> 
> **Expected behavior**
> In the example, the function should return:
> Desktop
> Documents

[...]

>  - Terminal: Xfce Terminal Emulator v0.8.7.4

I guess you have $TERM set to "xterm" or similar, but you are not
actually using an xterm.  The t_TI and t_TE termcap entries are then
wrong for you.  You'll have to clear them:
	set t_TI= t_TE=

I don't think we can do this automatically, because outputting these
codes was necessary to fix a problem.

-- 
hundred-and-one symptoms of being an internet addict:
162. You go outside and look for a brightness knob to turn down the sun.

 /// Bram Moolenaar -- Bram@Moolenaar.net -- http://www.Moolenaar.net   \\\
///        sponsor Vim, vote for features -- http://www.Vim.org/sponsor/ \\\
\\\  an exciting new programming language -- http://www.Zimbu.org        ///
 \\\            help me help AIDS victims -- http://ICCF-Holland.org    ///

**Comment by betheev** [comment]
Thanks for the prompt response. Your suggestion helped me solved the problem. I don't know when does $TERM get set nor by whom. I echoed its value on two different computers running Xubuntu 18.04.4 LTS and using the Xfce Terminal Emulator v0.8.7.4 and I got the same response: xterm-256color. Its possible that other Xubuntu users will also experience this problem once they upgrade their Vim version. I also run `:echo system("ls")` with vim v8.2.803 on an xterm terminal emulator without modifying t_TI and t_TE and didn't get the escape codes.
